### ISSUE TITLE
RDX: Remove image tag from extension ID

### ISSUE THREAD

**Issue by mook-as** [rancher-sandbox/rancher-desktop#4416]
### Problem Description

Currently, we use the full image reference (including the tag) for the extension ID. This will not work well with upgrades, and makes it possible to install two versions of the same extension concurrently.

### Proposed Solution

We should split the extension ID to exclude the tag, and add a new field to hold the tag.

#### Acceptance Criteria

- Try to install an extension ID specifying a tag. RDX must install the requested tag.
- Try to install an extension ID with a tag that does not exist in the registry. RDX must throw an error.
- Try to install an extension ID that is already installed with a different tag. RDX will automatically uninstall the existing tag and install the requested tag. Different tags will be stored in the container engine, but only the latter will be the active version in the UI.

**Comment by IsaSih** [comment]
I am reopening the issue, as I found a bug when it comes to trying to install an extension. with a tag that does not exist in the registry. The error message is given as expected, however, if there's already an existing installation of the same image, the extension is uninstalled.

```
rdctl extension ls                                            
Extension IDs

aquasec/trivy-docker-extension:0.4.6
docker/logs-explorer-extension:0.2.2
julianb90/tachometer:0.0.3
prakhar1989/dive-in:0.0.8
splatform/epinio-docker-desktop:0.0.19
rdctl extension install splatform/epinio-docker-desktop:0.0.29
The image splatform/epinio-docker-desktop:0.0.29 has invalid extension metadata
{"message":"422 Unprocessable Entity"}
rdctl extension ls                                            
Extension IDs

aquasec/trivy-docker-extension:0.4.6
docker/logs-explorer-extension:0.2.2
julianb90/tachometer:0.0.3
prakhar1989/dive-in:0.0.8
```

**Comment by mook-as** [comment]
Eric caught that reviewing #4658 so once that's merged we should be able to come back and double check that it's fixed.

**Comment by IsaSih** [comment]
Found a different issue on windows that do not match the acceptance criteria:

When trying to install an already existing extesnion but with a different tag, the extension is not being replaced by the new tag. It actually throws an error, and the extension ends up being completely uninstalled. As it is in the output below:

```
C:\Users\tuctuc>rdctl extension install splatform/epinio-docker-desktop:0.0.21
Installing image splatform/epinio-docker-desktop:0.0.21: Created

C:\Users\tuctuc>rdctl extension install splatform/epinio-docker-desktop:0.0.19
The image splatform/epinio-docker-desktop:0.0.19 failed to install: Could not copy host binary /windows/kubectl.exe
{"message":"422 Unprocessable Entity"}

C:\Users\tuctuc>rdctl extension ls
No extensions are installed.

```
Should I file a new issue for this, or do you prefer that I keep it here as a comment?

**Comment by mook-as** [comment]
I believe that's a distinct issue (probably with the nerdctl backend, but I'll test both), since it's related to installing an extension multiple times rather than the versions.  I'll make a PR separately (and try to make sure it's covered by tests).

**Comment by IsaSih** [comment]
This behavior is now addressed in #4722

**Comment by gaktive** [comment]
With #4722 now filed, @IsaSih please move this to Done and we can focus on that other ticket in an upcoming milestone.
